src/frontend/org/voltdb/exportclient/kafka/KafkaExportClient.java

```
@@ -295,22 +326,28 @@ public ExportDecoderBase constructExportDecoder(AdvertisedDataSource source) {
         String m_topic = null;
         boolean m_primed = false;
         Properties m_decoderProducerConfig;
-        KafkaProducer<String, String> m_producer;
-        final CSVStringDecoder m_decoder;
+        KafkaProducer<String, Object> m_producer;
+        final RowDecoder<?, RuntimeException> m_decoder;
         final List<Future<RecordMetadata>> m_futures = new ArrayList<>();
         private final AtomicBoolean m_failure = new AtomicBoolean(false);
         final ListeningExecutorService m_es;
 
         public KafkaExportDecoder(AdvertisedDataSource source) {
             super(source);
 
-            CSVStringDecoder.Builder builder = CSVStringDecoder.builder();
-            builder
-                .dateFormatter(Constants.ODBC_DATE_FORMAT_STRING)
-                .timeZone(m_timeZone)
-                .binaryEncoding(m_binaryEncoding)
-                .skipInternalFields(m_skipInternals)
-            ;
+            if (m_encodeFormat == EncodeFormat.AVRO) {
+                m_decoder = new AvroDecoder.Builder().build();
+            } else {
+                CSVStringDecoder.Builder builder = CSVStringDecoder.builder();
```

See previous comment: I think we must now test the format values explicitly and avoid silently defaulting to CSV.